src/public/modules/forms/admin/directives/configure-mobile.client.directive.js

```diff
@@ -10,24 +18,79 @@ function configureMobileDirective() {
       'modules/forms/admin/directiveViews/configure-mobile.client.view.html',
     restrict: 'E',
     scope: {
-      field: '=',
+      field: '<',
+      form: '<',
       name: '=',
       characterLimit: '=',
+      isLoading: '<',
     },
     controller: [
+      '$q',
       '$uibModal',
       '$scope',
       '$translate',
-      function ($uibModal, $scope, $translate) {
+      'Toastr',
+      function ($q, $uibModal, $scope, $translate, Toastr) {
         // Get support form link from translation json.
-        $translate('LINKS.SUPPORT_FORM_LINK').then((supportFormLink) => {
-          $scope.supportFormLink = supportFormLink
+        $translate('LINKS.TWILIO_SETUP_LINK').then((twilioSetupLink) => {
+          $scope.twilioSetupLink = twilioSetupLink
         })
 
+        // NOTE: This is set on scope as it is used by the UI to determine if the toggle is loading
+        $scope.isLoading = true
+        $scope.field.hasRetrievalError = false
+
+        const getAdminVerifiedSmsState = (verifiedSmsCount, msgSrvcId) => {
+          if (msgSrvcId) {
+            return ADMIN_VERIFIED_SMS_STATES.hasMessageServiceId
+          }
+          if (verifiedSmsCount < SMS_VERIFICATION_LIMIT) {
+            return ADMIN_VERIFIED_SMS_STATES.belowLimit
+          }
+          return ADMIN_VERIFIED_SMS_STATES.limitExceeded
+        }
+
+        $q.when(
+          AdminMetaService.getFreeSmsCountsUsedByFormAdmin($scope.form._id),
+        )
+          .then(({ smsCounts }) => {
+            $scope.verifiedSmsCount = smsCounts
+            $scope.adminVerifiedSmsState = getAdminVerifiedSmsState(
+              smsCounts,
+              $scope.form.msgSrvcName,
+            )
+            // NOTE: This links into the verifiable field component and hence, is used by both email and mobile
+            $scope.field.hasAdminExceededSmsLimit =
+              $scope.adminVerifiedSmsState ===
+              ADMIN_VERIFIED_SMS_STATES.limitExceeded
+            $scope.field.hasRetrievalError = false
```

Comment:
Probably don't need to set `hasRetrievalError` to `false` since its already set above?

Reply:
thanks for comment - changed!